eval-config.nix

```diff
@@ -1,10 +1,75 @@
+let
+  nixDarwinVersion = builtins.fromJSON (builtins.readFile ./version.json);
+
+  checkRelease = lib:
+    # Avoid breaking configurations when the unstable Nixpkgs version
+    # rolls over.
+    #
+    # TODO: Something more refined than this would be ideal, as this
+    # still means you could be using unstable nix-darwin 25.05 with
+    # Nixpkgs 26.05, which would be unfortunate.
+    if nixDarwinVersion.isReleaseBranch then
+      lib.trivial.release == nixDarwinVersion.release
+    else
+      lib.versionAtLeast lib.trivial.release nixDarwinVersion.release;
+in
+
 { lib
 , modules
 , baseModules ? import ./modules/module-list.nix
 , specialArgs ? { }
 , check ? true
+, enableNixpkgsReleaseCheck ? true
 }@args:
 
+assert enableNixpkgsReleaseCheck -> checkRelease lib || throw ''
+
+  nix-darwin now uses release branches that correspond to Nixpkgs
+  releases. You are currently using nix-darwin ${nixDarwinVersion.release} with
+  Nixpkgs ${lib.trivial.release}.
+
+  Check <https://status.nixos.org/> for the currently supported
+  releases of Nixpkgs. nix-darwin’s release support policy follows that
+  of Nixpkgs.
```

Comment:
```suggestion
  releases of Nixpkgs. nix-darwin only supports using the same version of Nixpkgs corresponding to the nix-darwin release.
```

Reply:
I was attempting to convey “we don’t support or update our release branches corresponding to unmaintained Nixpkgs branches”, but maybe that’s not really important to convey. If you want us to be more explicit about requiring the version match here, maybe I could strengthen the first paragraph of the error message?

Reply:
Yeah that would be good, I think for us we care more about making sure users aren't using mismatched versions, as the support surface for us is lower I don't mind backporting fixes to super old release branches for anyone who really wants to use 21.05 or something

Reply:
I’ve clarified the wording here and dropped the sentence about support; let me know what you think.

(We definitely shouldn’t cut branches for releases older than 24.11, I don’t think. Backporting down to 21.05 would mean going through seven stable releases along the way, unless we reserve some features only for specific ancient branches. That all sounds like a nightmare and I don’t plan to do any work for unmaintained Nixpkgs branches that become insecure and unsafe to use almost immediately, but I guess as long as I’m not doing the backports or replying to the support requests it doesn’t really matter to me.)